src/planet-a/Active/SysPlanetA.cpp

```diff
@@ -75,30 +74,34 @@ StatusActivated SysPlanetA::activate_sat(
     ActiveEnt root = scene.hier_get_root();
     ActiveEnt planetEnt = scene.hier_create_child(root, "Planet");
 
-    auto &planetTransform = scene.get_registry()
-                            .emplace<osp::active::ACompTransform>(planetEnt);
-    planetTransform.m_transform = Matrix4::translation(positionInScene);
+    auto &rPlanetTransform = scene.get_registry()
+                             .emplace<osp::active::ACompTransform>(planetEnt);
+    rPlanetTransform.m_transform = Matrix4::translation(positionInScene);
     scene.reg_emplace<ACompFloatingOrigin>(planetEnt);
 
-    auto &planetPlanet = scene.reg_emplace<ACompPlanet>(planetEnt);
-    planetPlanet.m_radius = loadMePlanet.m_radius;
+    auto &rPlanetPlanet = scene.reg_emplace<ACompPlanet>(planetEnt);
+    rPlanetPlanet.m_radius = loadMePlanet.m_radius;
+
+    auto &rPlanetForceField = scene.reg_emplace<ACompFFGravity>(planetEnt);
 
-    auto &planetForceField = scene.reg_emplace<ACompFFGravity>(planetEnt);
+    // gravitational constant
+    static const float sc_GravConst = 6.67408E-11f;
 
-    // arbitrarily picked nice working number
-    planetForceField.m_Gmass = 100000.0f;
+    rPlanetForceField.m_Gmass = loadMePlanet.m_mass * sc_GravConst;
 
     return {0, planetEnt, false};
 }
 
-SysPlanetA::SysPlanetA(osp::active::ActiveScene &scene) :
+SysPlanetA::SysPlanetA(osp::active::ActiveScene &scene,
+                       osp::UserInputHandler &userInput) :
     m_scene(scene),
     m_updateGeometry(scene.get_update_order(), "planet_geo", "", "physics",
-                     std::bind(&SysPlanetA::update_geometry, this)),
+                    [this] { this->update_geometry(); } ),
```

Comment:
Do we need to care about lifetime management of this?

Are we guaranteed that whatever m_updateGeometry talks to will always be dead before "this" is dead?

Reply:
The dynamic systems added to the scene stay with it throughout its entire lifetime. The scene does not call any updates during destruction.

Reply:
Put some documentation related to that somewhere ?